src/App.js

```diff
@@ -3,14 +3,41 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
+
 function App() {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsloading] = React.useState(true);
+
+  React.useEffect(() => new Promise((resolve,reject) => setTimeout( () => 
+  resolve({ data: {todoList: JSON.parse(localStorage.getItem("savedTodoList"))}}), 2000))
+  .then(function(result) {
+    console.log(result);
+    setTodoList(result.data.todoList);//is this correct?
+    setIsloading(false);//is this correct?
+  }),[]);
```

Comment:
OK, first of all, yes both lines where you commented with a question are correct. Perhaps discuss with your tutor what you weren't sure about, but I don't see anything wrong there. 

However the above block of code is quite difficult to read. I would structure it like this:
```
React.useEffect(() => {
  new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve({data: {
        todoList:JSON.parse(localStorage.getItem("savedTodoList"))
      }})
    }, 2000);
  })
  .then((result) => {
    setTodoList(result.data.todoList);
    setIsLoading(false);
  });
}, []);
```